### ISSUE TITLE
Using Go's native SSH implementation will not open interactive shell

### ISSUE THREAD

**Issue by gbraad** [minishift/minishift#1005]
## Steps to reproduce
```
$ sudo mv /usr/bin/ssh /usr/bin/-ssh
$ minishift start
$ minishift ssh
$ 
```
this behaviour is also observed in Minikube

while:
```
$ docker-machine start default
$ docker-machine ssh
# 
```

works.

**Comment by gbraad** [comment]
Notes:
  * https://github.com/docker/machine/blob/master/libmachine/host/host.go#L75
  * https://github.com/docker/machine/blob/e1a03348ad83d8e8adb19d696bc7bcfb18ccd770/commands/ssh.go#L18
  * https://github.com/kubernetes/minikube/blob/master/pkg/minikube/cluster/cluster.go#L537

**Comment by budhrg** [comment]
I had mentioned that earlier here https://github.com/minishift/minishift/pull/911#issuecomment-302043786. 

+1 for fixing it.

**Comment by gbraad** [comment]
Still not sure why the QnD approach is needed, as it seems https://github.com/docker/machine/blob/master/libmachine/ssh/client.go#L318 actually checks for the length of the args.

**Comment by gbraad** [comment]
@budhrg the understanding was that the native-client did not support interactive shell, but this is because we inherited it from Minikube. :-s Still looking for a cleaner fix.

**Comment by gbraad** [comment]
We invoke the shell as follows https://github.com/minishift/minishift/blob/master/pkg/minikube/cluster/cluster.go#L511
```go
	return client.Shell(strings.Join(args, " "))
```
What happens is that we provde an empty string instead of no value: https://play.golang.org/p/z7DDUWnuYm (you can verify this with:

```go
package main

import (
	"strings"
	"fmt"
	
)

func main() {
	args := []string {}
	command := strings.Join(args, " ")
	fmt.Printf("Type: %T, Length: %v, Value: %v.\n", command, len(command), command)
}
```

At https://github.com/docker/machine/blob/master/libmachine/ssh/client.go#L269 the use of `args ...string` will result in args having a length of 1.

```go
func (client *NativeClient) Shell(args ...string) error {
```

this means that the check at https://github.com/docker/machine/blob/master/libmachine/ssh/client.go#L318 wrongly assumes that arguments are given, and it will execute https://github.com/docker/machine/blob/master/libmachine/ssh/client.go#L324

```go
		session.Run(strings.Join(args, " "))
```

Docker machine itself uses a different way to invoke the shell https://github.com/docker/machine/blob/e1a03348ad83d8e8adb19d696bc7bcfb18ccd770/commands/ssh.go#L50

```go
	return client.Shell(c.Args().Tail()...)
```

**Comment by budhrg** [comment]
> We invoke the shell as follows  ....

Thanks for the wonderful explaination @gbraad Appreciated!

**Comment by hferentschik** [comment]
Merged via pull request #1009
